lib/grpcServer/handlers/tx-filter-stream/subscribeToTransactionsWithProofsHandlerFactory.js

```diff
@@ -164,17 +155,51 @@ function subscribeToTransactionsWithProofsHandlerFactory(
       );
     }
 
+    // If block height is specified instead of block hash, we obtain block hash by block height
+    if (fromBlockHash === '') {
+      if (fromBlockHeight === 0) {
+        throw new InvalidArgumentGrpcError('minimum value for `fromBlockHeight` is 1');
+      }
+
+      // we don't need to check bestBlockHeight because getBlockHash throws
+      // an error in case of wrong height
+      try {
+        fromBlockHash = await coreAPI.getBlockHash(fromBlockHeight);
+      } catch (e) {
+        if (e.code === -8) {
+          // Block height out of range
+          throw new InvalidArgumentGrpcError('fromBlockHeight is bigger than block count');
+        }
+
+        throw e;
+      }
+    }
+
+    // Send historical transactions
+
+    const fromBlock = await coreAPI.getBlock(fromBlockHash);
```

Comment:
We need to handle error if block hash was incorretly set. Also, if it possible we need to check that block is in the right (best) chain.

Reply:
What do you mean by "incorrectly set"? Can you give an example of an incorrect value? Also, you want to check that the hash user has submitted is on the best chain? I mean, why would you do that? The best chain at this moment in time doesn't mean that it will always be the best chain, also a chain with less work can in theory overtake the best chain at the moment. Does this check make any sense here?

Reply:
I mean user just passed an incorrect block hash that is not present in the chain. Also, if a user pass a hash of the block which is not part of the chain then `getHistoricalTransactions` method won't work because there we are getting height by hash so it won't work properly.